playground/yarp/Yarp.AppHost/Program.cs

```diff
@@ -25,6 +25,10 @@
                                    .WithTransformPathRemovePrefix("/api");
                            });
 
+#pragma warning disable ASPIREHOSTING001 // Type is for evaluation purposes only
+var viteApp = builder.AddYarpNpmApp("vite-app", Path.Combine("..", "vite-app"));
```

Comment:
I think this is ultimately an example of a resource that would be capable of emitting steps in the pipeline. A lot of the state encapsulated in this resource is basically the command to run (install/build) and the state it needs (output path, input directory).

I wonder if we also need to be mindful of scenarios where the deployment takes certain shapes. For example, we have some users that need to install/build _outside_ of a Dockerfile then copy the resources into Docker.

Reply:
Yeah, I don't have a blocking objection to the changes here but my mind was definitely seeing that "npm install" and "npm build" are steps here and Docker is the pipeline runner that is being used to execute those steps.

We might be able to evolve this into the pipeline model when we get there.